File: scripts/Light_service.cpp
/*#include <ros/ros.h>
#include <ethernet_remote_io_module/WriteCoilsList.h>
#include <std_msgs/Int64.h>
#include <ethernet_remote_io_module/ReadDigitalInputs.h>


//using namespace std;


class LightControl_cpp {
    public:
    bool ctrl_c = false;
    ros::Publisher coils_pub;
    ros::Subscriber read_inputs_subscriber;
    
    public:
    LightControl_cpp(ros::NodeHandle *nh) {
        ethernet_remote_io_module::WriteCoilsList light;
        coils_pub = nh->advertise<ethernet_remote_io_module::WriteCoilsList>("write_coils", 10);    
        read_inputs_subscriber = nh->subscribe("read_inputs", 1000, &LightControl_cpp::read_digital_callback, this);

    }
    ethernet_remote_io_module::ReadDigitalInputs read_digital_callback(const ethernet_remote_io_module::ReadDigitalInputs& msg) {
        ethernet_remote_io_module::ReadDigitalInputs data;
        data = msg.data;
        return data;
        
    }
    void write_coils(const ethernet_remote_io_module::WriteCoilsList& msg)
    {
        while(ros::ok)
        {
            coils_pub.publish(msg);
        } 

    }

    void light_off()
    {
        light = [0,0,0,0,0,0,0]
        LightControl_cpp::write_coils(light)
    }

    void light_type_1_cpp()
    {
        light = [1,0,0,0,0,0,0]
        LightControl_cpp::write_coils(light)
    }
};

    

class Light_Service
{
    public:
          bool light_command(camel_robot::light::Request& req,
                            camel_robot::light::Response& res);
          
       
};


bool Light_Service::light_command(camel_robot::light::Request& req,
                                    camel_robot::light::Response& res)
{
    
    LightControl_cpp l;
    
    if(req.request_string == "light_off")
    {
        l.light_off()
        res.response_string = "ALL Lights are turned OFF!";

    }

    else if(req.request_string == "light_type_1")
    {
        //l::light_type_1();
        res.response_string = "Lighting_type_1 is on!";
    }

    else if(req.request_string == "light_type_2")
    {
        //l::light_type_2();
        res.response_string = "Lighting_type_2 is on!";
    }

    else if(req.request_string == "light_type_3")
    {
        //l::light_type_3();
        res.response_string = "Lighting_type_3 is on!";
    }

    else
    {
        res.response_string = "Undefined request String";
    }

    return true;

}

int main(int argc, char **argv)
{
  ros::init(argc, argv, "Light_Service_cpp");
  ros::NodeHandle n;
  ros::NodeHandle nh;
  LightControl_cpp nc = LightControl_cpp(&n);
  
  Light_Service a;

  
  ros::Publisher write_coils_publisher = n.advertise<ethernet_remote_io_module::WriteCoilsList>("write_coils", 10);
  ros::ServiceServer ss = n.advertiseService("Light_server", &Light_Service::light_command, &a);
// %EndTag(SERVICE_SERVER)%

  ros::spin();

  return 0;
}

*/
#include <iostream>

int main(){

    return 0;
}

File: scripts/Light_service.py
#!/usr/bin/env python3
from Light_control import LightControl
import rospy
from rospy.core import rospyerr ,loginfo
from camel_robot.srv import light
import time
class LightService:

    def __init__(self):
        self._as = rospy.Service('Light_server', light, handler=self.light_command)
        self.lc = LightControl()
        self.rate = rospy.Rate(10)    
        self.result = light()

    def light_command(self, command):
        self.command = command.request_string
        
        if(self.command == "light_off"):
            self.lc.light_off()
            loginfo("All Lights are turned OFF!")
            self.result = "ALL Lights are turned OFF!"
        
        elif(self.command == "light_type_1"):
            self.lc.light_type_1()
            self.result = "Lighting_type_1 is on!"
        
        elif(self.command == "light_type_2"):
            self.lc.light_type_2()
            self.result = "Lighting_type_2 is on!"
        
        elif(self.command == "light_type_3"):
            self.lc.light_type_3()
            self.result = "Lighting_type_3 is on!"
        
        elif(self.command == "light_type_4"):
            self.lc.light_type_4()
            self.result = "Lighting_type_4 is on!"
        
        elif(self.command == "light_type_5"):
            self.lc.light_type_5()
            self.result = "Lighting_type_5 is on!"
        
        elif(self.command == "light_type_6"):
            self.lc.light_type_4()
            # time.sleep(0.5)
            # self.lc.light_off()
            # time.sleep(0.5)
            # self.result = "Lighting_type_6 is on!"
        
        elif(self.command == "light_type_7"):
            self.lc.light_type_7()
            self.result = "Lighting_type_7 is on!"

        elif(self.command == "light_dance"):
            self.lc.light_dance()
            self.result = "Light Dance has been done!"
        
        else:
            loginfo("Undefined request String")
            self.result = "Undefined request String"
        return self.result
        
if __name__ == '__main__':
    rospy.init_node('Light_Service')

    server = LightService()

    rospy.spin()